app/controllers/problem_reports_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class ProblemReportsController < ApplicationController
+  def index
+    @problem_reports = ProblemReport.order(id: :desc)
+    filter_encode
+
+    if params[:filter] == 'Unfixed and Critical'
+      @problem_reports.where!(is_critical: true, is_fixed: false)
+    elsif params[:filter] == 'Unfixed and Fixable'
+      @problem_reports.where!(is_fixable: true, is_fixed: false)
+    end
+  end
+
+  def new; end
+
+  def create
+    create_new_report
+    if @report.save
+      flash[:notice] = 'Success'
+      redirect_to problem_reports_path
+    else
+      flash[:notice] = 'Fail'
+      redirect_to new_problem_report_path
+    end
+  end
+
+  def update
+    update_remarks
+    %w[is_fixable is_fixed is_blocked is_critical].each do |a|
+      if params[a]
+        val = !@report.public_send(a)
+        @report.public_send("#{a}=", val)
+      end
+    end
+
+    @report.save
+    redirect_to problem_reports_path
+  end
+
+  def update_remarks
+    @report = ProblemReport.find(params[:id])
+    @report.last_update_user_id = current_user.id
+    @report.remarks = params[:remarks] if params[:remarks]
+  end
+
+  private
+
+  def filter_encode
+    params[:filter] = if !params[:filter]
+                        'Unfixed and Fixable'
+                      else
+                        params[:filter][5..-10]
+                      end
+  end
+
+  def create_new_report
+    @report = ProblemReport.new
```

Comment:
`ProblemReport.new(reporter_user: current_user, last_update_user: current_user, ...)`

Reply:
Also, since this is only used in `create`, do not modify the instance variable `@report`. What you can do instead is to make this function return a Report